src/markup/codemirror/create.ts

```diff
@@ -152,7 +153,38 @@ export function createCodemirror(params: CreateCodemirrorParams) {
                 onScroll(event);
             },
             paste(event, editor) {
-                if (event.clipboardData && parseInsertedUrlAsImage) {
+                if (!event.clipboardData) return;
+
+                // Note: I have editor.dispatch() in the try-catch on purpose.
+                // The code's pretty new and there might be random issues we haven't caught yet,
+                // especially with invalid HTML or weird DOM parsing errors.
+                // If something goes wrong, I just want to fall back to the "default pasting"
+                // rather than break the entire experience for the user.
+                // It’s kind of like a temporary safety net right now until things are more stable.
+                try {
+                    // Handle HTML insertion
+                    const htmlContent = event.clipboardData.getData(DataTransferType.Html);
+                    if (htmlContent) {
+                        const parser = new DOMParser();
+                        const doc = parser.parseFromString(htmlContent, 'text/html');
+                        const links = Array.from(doc.getElementsByTagName('a'));
+
+                        if (links.length > 0) {
```

Comment:
Let's insert markdown with any copy from HTML/DOM? Not only if links ware copied

Reply:
fixed